tools/pd-simulator/simulator/simutil/transfer_region_counter.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simutil
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// TransferRegionCount is to count transfer schedule for judging whether redundant
+type TransferRegionCount struct {
+	storeNum        int
+	regionNum       int
+	IsValid         bool
+	Redundant       uint64
+	Necessary       uint64
+	regionMap       map[uint64]uint64
+	visited         []bool
+	GraphMat        [][]uint64
+	mutex           sync.Mutex
+	loopResultPath  [][]int
+	loopResultCount []uint64
+}
+
+// TransferRegionCounter is an global instance for TransferRegionCount
+var TransferRegionCounter TransferRegionCount
+
+// Init for TransferRegionCount
+func (c *TransferRegionCount) Init() {
+	c.storeNum = CaseConfigure.StoreNum
+	c.regionNum = CaseConfigure.RegionNum
```

Comment:
Is this mean total region number or a store region number?

Reply:
A store region number. It is defined by flag. https://github.com/lhy1024/pd/blob/9d88133030e2263be6b4eaa55c58e4f2857f1146/tools/pd-simulator/main.go#L49